### ISSUE TITLE
Deno 1.9.0, Uncaught InvalidData: invalid url scheme

### ISSUE THREAD

**Issue by FSou1** [denoland/deno_std#955]
I have recently published a module: https://deno.land/x/fsquery/mod.ts

It works locally and at [github actions](https://github.com/FSou1/fsquery/actions/runs/746091878).

Now I'm trying to test it:

```
import { fsselect } from 'https://deno.land/x/fsquery@deno-module-0.1/mod.ts';

await fsselect("select * from .");
```

Unfortunately, I have been receiving the error:
```
error: Uncaught InvalidData: invalid url scheme
    throw new Deno.errors.InvalidData("invalid url scheme");
          ^
    at fileURLToPath (https://deno.land/std@0.92.0/node/url.ts:53:11)
    at createRequire (https://deno.land/std@0.92.0/node/module.ts:535:18)
    at https://deno.land/x/fsquery@deno-module-0.1/parser.ts:3:17
```

Here is a tricky point, which actually causes this error:
```
import { createRequire } from "./deps.ts";
const require = createRequire(import.meta.url);
const parser = require("./peg-parser.js");
export default parser;
```

Update: I've changed the module declaration from CommonJS to ES6 module. The issue is resolved now, but still reproducible at the `fsquery@deno-module-0.1` version.

**Comment by TotallyNotChase** [comment]
@lucacasonato I'd like to take this on. What do you think the error message here should be? I think, ideally, it should highlight that the `filename` url should be of `file` protocol. But it could also include a note about how `createRequire` is expected to be used from `file` modules (though this note is only relevant when the passed argument is `import.meta.url`)

**Comment by lucacasonato** [comment]
Probably something along the lines of `createRequire only supports 'file://' URLs` would be enough.

**Comment by Haprog** [comment]
It seems like the old error message is still being used for this even in Deno 1.23.3. I ran into this when trying to use:

```ts
import { create } from 'https://deno.land/x/deno_exif@0.0.2/mod.ts';
```

> `createRequire` can only be used from `file` modules. We should improve the error message here.

First few times I was reading this I wasn't even sure what this means. First I thought what even is a "file module"? And what is the case that is not working/supported then?

Do I understand it correctly now that `createRequire` does not work for importing via HTTP/HTTPS URLs?

If so, this should definitely be clarified in the documentation (at least [here](https://deno.land/manual/node/std_node#loading-commonjs-modules) and [here](https://doc.deno.land/https://deno.land/std@0.147.0/node/module.ts/~/default#createRequire)).

Basically any code that uses `createRequire` for example:
```ts
const require = createRequire(import.meta.url);
const something = require("./something");
```
does not work when imported via HTTP/HTTPS URLs.

**Comment by kt3k** [comment]
@Haprog Thanks for reporting. Right, the error message doesn't look great now. I opened #2440 to try to address this
